### ISSUE TITLE
What license is cool-retro-term under?

### ISSUE THREAD

**Issue by nadams810** [Swordfish90/cool-retro-term#269]
I see a file for GPL 2.0 and GPL 3.0. I suggest you delete one and specify the license in the README to avoid confusion.

**Comment by KAMiKAZOW** [comment]
Some files are under GPLv2. Deleting the license text would be illegal.
The About window clearly says that the overall project is under GPLv3.

**Comment by nadams810** [comment]
http://gplv3.fsf.org/rms-why.html

    When we say that GPLv2 and GPLv3 are incompatible, it means there is no legal way to combine code under GPLv2 with code under GPLv3 in a single program.

That is by Richard Stallman himself. This is one of the reasons why I'm not a fan of the GPL....but I digress. So if this project is GPLv3 - you need to remove GPLv2 code. Or change the license to GPLv2.

    The About window clearly says that the overall project is under GPLv3.

I shouldn't have to compile the program to find out what the license the software falls under.

As stated in the book by Karl Fogel (who is one of the creators of Subversion) - Producing Open Source Software

http://producingoss.com/en/license-quickstart.html

    The first thing to do is state the license clearly on the project's front page.

**Comment by KAMiKAZOW** [comment]
The GPLv2 code is not GPLv2-only. It's GPLv2 or any later version.
Read the headers and don't assume made-up BS.

**Comment by nadams810** [comment]
If you disagree with what Karl Fogel feels an open source project should do - please do contact him. I'm pretty sure many people would agree with his opinions and even though they didn't read his book - follows what he suggests.

The "made up BS" was from your comment - there was no other way to interpret that:

    Some files are under GPLv2

Which the linked project also does not specify a license in their README.

Stating that a developer should comb through every file looking for licenses doesn't seem reasonable. Especially if it is a larger project spanning hundreds if not thousands of files (The [Doom 3 source](https://github.com/TTimo/doom3.gpl) is a perfect example of this).

Please don't take offense to this - you can do whatever you want. This was meant to be a friendly suggestion to make what license this project is released under clear. If you feel that telling people what license this project is released under (as well as included projects) is a non-issue then I will "close and comment" this issue.

**Comment by Swordfish90** [comment]
Hi @nadams810 and sorry for the late response. Since the project is split into two different components, the idea behind the licensing was the following:

1) All the files belonging to QMLTermWidget which is a separate library forked by qtermwidget are licensed under GPL2 (as the original library).
2) All the files belonging to CRT are licensed under GPL3.

To be fair I'm not really an expert when it comes to licensing and if there is something wrong with the following approach just let me know.

**Comment by KAMiKAZOW** [comment]
He's not an expert either. He just acts as if he's one. You're doing nothing wrong, @Swordfish90.

**Comment by nadams810** [comment]
@Swordfish90 Including multiple licenses and not specifying which one CRT is licensed under is rather confusing. This is only due to the fact that if I wanted to use code just from your CRT application - which license do I use?  It does appear that @KAMiKAZOW disagrees with what many open source developers do today. Ultimately it's your project - do whatever you want. I already offered to close it if @KAMiKAZOW felt this wasn't really an issue but got no response.

**Comment by KAMiKAZOW** [comment]
CRT's license is specified: https://github.com/Swordfish90/cool-retro-term/blob/master/app/qml/AboutDialog.qml
Don't make shit up!

**Comment by nadams810** [comment]
I apologize that if you believe I am making stuff up. I think there must be a language barrier.

In any case there are many, popular, projects that clearly define their licenses upfront:

https://github.com/alfajango/jquery-dynatable
https://github.com/TTimo/doom3.gpl
https://github.com/box/spout
https://github.com/PHPOffice/PHPExcel
https://github.com/PyMySQL/PyMySQL
https://github.com/farbrausch/fr_public
https://github.com/git/git
https://github.com/sumatrapdfreader/sumatrapdf
https://github.com/notepad-plus-plus/notepad-plus-plus

I'm going to mark this as "close and comment" because @KAMiKAZOW appears unwilling to discuss any opinion on how licenses should be displayed to users and developers. Which is disappointing really - I always want to hear constructive opinions brought to the table because that is how methodologies and processes improve. If we didn't - we would still be programming with punch cards (or worse).

**Comment by dcrockwell** [comment]
+1 disappointing KAMiKAZOW derailed this conversation.

**Comment by Luke-Nukem** [comment]
I have to agree with @nadams810 that it would be very helpful to list which licences applying to which code collections up front. I notice some files are also missing their license headers :smiley_cat:

**Comment by nadams810** [comment]
I believe there is a consensus that this is really a problem so I'll reopen and see what @Swordfish90 wants to do.

**Comment by KAMiKAZOW** [comment]
https://github.com/Swordfish90/cool-retro-term/blob/master/app/qml/AboutDialog.qml#L92

Not that hard to find. I gave the same answer two years ago already.

**Comment by akien-mga** [comment]
And as answered above, that's not the recommended way to apply the GPL-3.0 to source code. But I rest my case, if the final answer is EDONTCARE, I guess @nadams810 can close it again and you can lock it so that people with some licensing knowhow stop whining every year or so :)

**Comment by KAMiKAZOW** [comment]
People who "whine" (as you put it) about the issue, could send a pull request for a changed Readme with the license declaration from AboutDialog.qml pasted in. I'm sure Filippo would accept it.

**Comment by akien-mga** [comment]
First of all sorry for my sarcastic tone above.

> could send a pull request for a changed Readme with the license declaration from AboutDialog.qml pasted in.

That's the point where my experience differs, and I've been around open source software for a while. If repository maintainers are not open to a given change, it's a waste of time to implement it, make a pull request, and see it reverted.

So far this issue has not shown any willingness to make the situation clearer (which would imply a statement in the README.md, and ideally the addition of the *right* copyright headers to all code files, since some are GPL-3.0 and others are GPL-2.0 apparently), which is why I asked for a status update.

**Comment by Luke-Nukem** [comment]
It shouldn't be up to a random contributor to try and clarify a license issue - after-all they didn't create the product and they don't know the authors intentions.

**Comment by KAMiKAZOW** [comment]
There is no licensing issue. The license is properly documented in the "About" window.

**Comment by Luke-Nukem** [comment]
> There is no licensing issue. The license is properly documented in the "About" window.

There's a *big* licensing issue. There are two license files in the root, and it is not clear which is the correct license - the readme doesn't tell you, and the source files don't have a license header. 

> To do so, attach the following notices to the program. It is safest to attach them to the start of each source file to most effectively state the exclusion of warranty; and each file should have at least the “copyright” line and a pointer to where the full notice is found.

If the above were done, then this would have been a non-issue from the start.

And now you're telling us that *once the program is compiled it tells you which license is correct?*.

Mate, that's a problem. And one that could be solved by the author very quickly and simply by stating exactly which license applies to the entire written source in the readme. And could be made concrete by including a license header in all applicable source files.

**Comment by akien-mga** [comment]
I don't understand why there is no interest in fixing this issue? Finding out the exact licensing terms shouldn't be so hard.

I actually landed here again thinking "heh, I thought I had packaged cool-retro-term for my distro, which can't I find it?" and deciding to give it a try again... to bump into the same lack of clear license statement which likely made me drop the idea 2 years ago.
